activerecord/lib/active_record/railties/query_log_tags.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Railties # :nodoc:
+    module QueryLogTags #:nodoc:
+      module ActionController
+        extend ActiveSupport::Concern
+
+        included do
+          if ::ActionController::Base.query_log_tags_action_filter_enabled
+            around_action :record_query_log_tags
+          end
+          ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryLogTagsContext.extend(ControllerContext)
+        end
+
+        def record_query_log_tags
+          ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryLogTagsContext.update(controller: self)
+          yield
+        ensure
+          ActiveRecord::ConnectionAdapters::AbstractAdapter::QueryLogTagsContext.update(controller: nil)
+        end
+
+        module ControllerContext # :nodoc:
+          def controller
+            context[:controller]&.controller_name
+          end
+
+          def controller_with_namespace
+            context[:controller]&.class&.name
+          end
+
+          def action
+            context[:controller]&.action_name
+          end
+        end
+      end
+
+      module ActiveJob
```

Comment:
is it okay to reference `ActiveJob` inside `ActiveRecord`?

Reply:
It's a tricky one - it's not really referencing the other projects until these railties are called in by those other projects. It's more of a staging area and logical grouping.

Reply:
Resolving as these railties have been removed :)